checker/src/org/checkerframework/checker/index/upperbound/OffsetEquation.java

```diff
@@ -245,6 +245,65 @@ public boolean isNegativeOrZero() {
         return isInt() && getInt() <= 0;
     }
 
+    /**
+     * Evaluates an offset term. If the term is an integer constant, returns its value. Otherwise,
+     * returns null.
+     */
+    private Integer evalConstantTerm(Receiver termReceiver) {
+        if (termReceiver instanceof FlowExpressions.ValueLiteral) {
+            // Integer literal
+            Object value = ((FlowExpressions.ValueLiteral) termReceiver).getValue();
+            if (value instanceof Integer) {
+                return (Integer) value;
+            }
+        } else if (termReceiver instanceof FlowExpressions.MethodCall) {
```

Comment:
I don't like that this is specialized to string length, but I'll accept it for now. Can you add a TODO to generalize this here before you merge?